azurelinuxagent/ga/exthandlers.py

```diff
@@ -988,8 +1071,8 @@ def report_ext_handler_status(self, vm_status, ext_handler, goal_state_changed):
         # For MultiConfig, we need to report status per extension even for Handler level failures.
         # If we have HandlerStatus for a MultiConfig handler and GS is requesting for it, we would report status per
         # extension even if HandlerState == NotInstalled (Sample scenario: ExtensionsGoalStateError, DecideVersionError, etc)
-        # We also need to report extension status for an uninstalled handler if extensions are disabled because CRP
-        # waits for extension runtime status before failing the extension operation.
+        # We also need to report extension status for an uninstalled handler if extensions are disabled, or if the extension
+        # failed due to policy, because CRP waits for extension runtime status before failing the extension operation.
```

Comment:
remove this comment - we are no longer reporting extension status if failed due to policy

Reply:
We treat policy like other handler-level failures - handler status with terminal code is sufficient for CRP to fail fast. 

We still create the ext status file in __report_policy_error() so that collect_ext_status() does not create a dummy "transitioning" status (which causes single-config extensions to timeout). But we don't enter the "if" condition here.